### ISSUE TITLE
Bug: utilsScriptDeferred can fire before utilsLoaded

### ISSUE THREAD

**Issue by nlwillia** [Bluefieldscom/intl-tel-input#303]
This is hard to explain, but should be easy to fix.  If multiple widgets are initialized separately and the `utilsScript` option is used, then it's possible for a caller to observe the resolution of `utilsScriptDeferred` before the `utilsLoaded` function has been fired by the ajax success hook.  This can lead to client code running before a widget is fully initialized.

The problem is how the conditions in `loadUtils` are structured.  `loadUtils` is called once per widget instance, so the ajax call is gated behind the static `loadedUtilsScript` flag.  But once that flag is set, control currently just falls through to the else which immediately resolves `utilsScriptDeferred` whether the ajax call has finished or not.  Callers who use that deferred to block until initialization has finished can end up executing too soon.

To be fully deterministic, the logic in `loadUtils` should be:
````
if (utilsScript) {
  if (!$.fn[pluginName].loadedUtilsScript) {
    $.fn[pluginName].loadedUtilsScript = true;
    $.ajax(...);
  }
} else {
  this.utilsScriptDeferred.resolve();
}
````

Here's a [codepen demo](http://codepen.io/anon/pen/xwgRZz?editors=101) showing how this can cause incorrect behavior.

**Comment by jackocnr** [comment]
Thanks for pointing this out - yes it looks like a bug. I don't have time to look into this, but I'm open to pull requests!

**Comment by jackocnr** [comment]
Fixed in v6.2.1

**Comment by nlwillia** [comment]
Terrific, thanks!
